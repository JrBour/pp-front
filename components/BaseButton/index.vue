<template>
  <button
    :type="type"
    :disabled="disabled"
    :class="{ reduce: small }"
    :style="{ padding: loading ? '3px 0' : '1em 0' }"
    @click="handleClick"
  >
    <Loader v-if="loading" opacity="1" fill="#fff" height="50px" />
    <span v-else>{{ text }}</span>
  </button>
</template>

<script>
export default {
  props: {
    loading: {
      type: Boolean,
      default: false
    },
    disabled: {
      type: Boolean
    },
    type: {
      type: String,
      default: 'button'
    },
    small: {
      type: Boolean,
      default: false
    },
    text: {
      type: String,
      required: true
    }
  },
  methods: {
    handleClick() {
      this.$emit('handle-click')
    }
  }
}
</script>

<style lang="scss" scoped>
button {
  box-sizing: content-box;
  width: 100%;
  border: none;
  border-radius: 10px;
  background: #3750b2;
  color: white;
  padding: 1em 0;
  font-weight: 500;
  font-size: 1.2em;
  margin: 4vh 0 2vh 0;

  &:disabled {
    opacity: 0.5;
  }
}
.reduce {
  display: block;
  width: 80%;
  margin: 3vh auto;
  font-size: 0.9em;
}
</style>
