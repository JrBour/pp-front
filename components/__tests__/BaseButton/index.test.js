import { shallowMount } from '@vue/test-utils'
import BaseButton from '@/components/BaseButton'

describe('BaseButton', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(BaseButton, {
      propsData: {
        text: 'button'
      },
      stubs: {
        Loader: true
      }
    })
  })

  // describe('methods', () => {
  //   it('should emit the method on-click', async () => {
  //     const button = wrapper.find('button')
  //     await button.trigger('click')
  //     expect(wrapper.emitted('on-click')).toBeTruthy()
  //   })
  //   it('should change the return of the computed method for button with primary theme', () => {
  //     wrapper.setData({ hover: true })
  //     expect(wrapper.vm.colorArrow).toBe('blue')
  //     wrapper.setData({ hover: false })
  //     expect(wrapper.vm.colorArrow).toBe('white')
  //     wrapper.setProps({ disabled: true })
  //     expect(wrapper.vm.colorArrow).toBe('gray')
  //   })
  //   it('should change the return of the computed method for button with secondary theme', () => {
  //     wrapper.setProps({ theme: 'secondary' })
  //     wrapper.setData({ hover: true })
  //     expect(wrapper.vm.colorArrow).toBe('white')
  //     wrapper.setData({ hover: false })
  //     expect(wrapper.vm.colorArrow).toBe('blue')
  //   })
  // })

  describe('snapshot', () => {
    it('renders properly', () => {
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
