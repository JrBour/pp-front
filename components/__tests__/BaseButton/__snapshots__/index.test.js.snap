// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton snapshot renders properly 1`] = `<button type="button" class="" style="padding: 1em 0px;"><span>button</span></button>`;
